docs/reference/api/storage/LittleFileSystem.md

```diff
@@ -1,11 +1,60 @@
 ## LittleFileSystem
 
-[Add description here.]
+The little filesystem (LittleFS) is a fail-safe filesystem designed for embedded systems, specifically for microcontrollers that use flash storage.
+
+```
+   | | |     .---._____
+  .-----.   |          |
+--|o    |---| LittleFS |
+--|     |---|          |
+  '-----'   '----------'
+   | | |
+```
+
+Microcontrollers and flash storage present three challenges for embedded storage: [power loss](#power-loss-resilience), [wear](#wear-leveling) and [limited RAM and ROM](#bounded-ram-and-rom). The LittleFS provides a solution to all three of these problems.
+
+- **Bounded RAM/ROM** - The LittleFS works with a limited amount of memory. The LittleFS avoids recursion and limits dynamic memory to configurable buffers that can be provided statically.
+
+- **Power-loss resilient** - The LittleFS is designed for operating systems that may have random power failures. The LittleFS has strong copy-on-write guarantees and keeps storage on disk in a valid state.
+
+- **Wear leveling** - Because the most common form of embedded storage is erodible flash memories, the LittleFS provides a form of dynamic wear leveling for systems that cannot fit a full flash translation layer.
+
+### When to use the LittleFS
+
+The LittleFS is intended for microcontrollers with external flash storage. In this context, the LittleFS outperforms the other Mbed OS file systems in terms of RAM, ROM, wear and runtime.
+
+For storage on an SD card that is accessible from a PC, use the FATFileSystem due to its portability.
+
+For internal flash, the LittleFS is compatible with the <a href="https://github.com/ARMmbed/flashiap-driver" target="_blank">flash IAP driver</a>, and developers have used it successfully in several projects. However, in size and erase cycles, internal flash is more expensive than external flash, and internal flash comes with the cost of locking up the system during erase cycles. For these reasons, we discourage internal flash for general storage use.
```

Comment:
How does this work not being a BlockDevice? This whole statement could go away by saying its for external NOR flash above in the intro

Reply:
The flashiap driver is a block device [I believe](https://github.com/ARMmbed/flashiap-driver/blob/master/FlashIAPBlockDevice.h#L40). Is that what you were referring to not being a block device?

Reply:
We intentionally didn't do that in mbed os because of the intrusive implications. Fine as an external component but wouldn't mention or promote from our docs. It gives the wrong impression

Reply:
This paragraph documents the reasons to not use Flash IAP. Otherwise we have a bunch of users asking how to use LittleFS with IAP, which is what we have been seeing.